tests/test_queue.py

```diff
@@ -0,0 +1,221 @@
+            use_title=config['model_params'].get('use_title'),
+            use_abstract=config['model_params'].get('use_abstract'),
+            use_cuda=config['model_params'].get('use_cuda'),
+            batch_size=config['model_params'].get('batch_size'),
+            knn=config['model_params'].get('knn'),
+            sparse_value=config['model_params'].get('sparse_value')
+        )
+        elmoModel.set_archives_dataset(archives_dataset)
+        elmoModel.set_submissions_dataset(submissions_dataset)
+        return elmoModel
+    return simple_elmo
+
+@pytest.fixture()
+def openreview_context():
+    """
+    A pytest fixture for setting up a clean expertise-api test instance:
+    `scope` argument is set to 'function', so each function will get a clean test instance.
+    """
+    config = {
+        "LOG_FILE": "pytest.log",
+        "OPENREVIEW_USERNAME": "openreview.net",
+        "OPENREVIEW_PASSWORD": "1234",
+        "OPENREVIEW_BASEURL": "http://localhost:3000",
+        "SUPERUSER_FIRSTNAME": "Super",
+        "SUPERUSER_LASTNAME": "User",
+        "SUPERUSER_TILDE_ID": "~Super_User1",
+        "SUPERUSER_EMAIL": "info@openreview.net",
+        "SPECTER_DIR": '../expertise-utils/specter/',
+        "MFR_VOCAB_DIR": '../expertise-utils/multifacet_recommender/feature_vocab_file',
+        "MFR_CHECKPOINT_DIR": '../expertise-utils/multifacet_recommender/mfr_model_checkpoint/',
+        "WORKING_DIR": 'TEMP_DIR',
+        "TEST_NUM": random.randint(1, 100000)
+    }
+    app = expertise.service.create_app(
+        config=config
+    )
+
+    with app.app_context():
+        yield {
+            "app": app,
+            "test_client": app.test_client(),
+            "config": config
+        }
+
+@pytest.fixture(scope="session")
+def celery_config():
+    return {
+        "broker_url": "redis://localhost:6379/0",
+        "result_backend": "redis://localhost:6379/0",
+        "task_track_started": True,
+        "task_serializer": "pickle",
+        "result_serializer": "pickle",
+        "accept_content": ["pickle", "application/x-python-serialize"],
+        "task_create_missing_queues": True,
+    }
+
+@pytest.fixture(scope="session")
+def celery_includes():
+    return ["expertise.service.celery_tasks"]
+
+@pytest.fixture(scope="session")
+def celery_worker_parameters():
+    return {
+        "queues": ("userpaper", "expertise"),
+        "perform_ping_check": False,
+        "concurrency": 4,
+    }
+
+def test_elmo_queue(openreview_context, celery_app, celery_worker):
+    test_client = openreview_context['test_client']
+    server_config = openreview_context['config']
+    test_num = server_config['TEST_NUM']
+    
+    if os.path.isdir(f'~{test_num}'):
+        shutil.rmtree(f'~{test_num}')
+
+    # Gather config
+    config = {
+        'name': 'test_run',
+        'match_group': ["ABC.cc"],
+        'dataset': {
+            'directory': 'tests/data/'
+        },
+        'csv_submissions': 'csv_submissions.csv',
```

Comment:
`csv_submissions.csv` is used because there does not exist an invitation in the mock client that has corresponding submissions - so I copy over the data from the test directory into the job directory. 
Currently it's implemented as a feature of the API in test mode - the key `csv_submissions` gets overwritten to contain a list of CSV strings and those get written to the server.

I could look into making this done exclusively by the tests, or looking into inserting submissions + a submission invitation into the mock client.